pages/blog/_posts/2021-08-01-the-specs-behind-the-specs.md

```diff
@@ -0,0 +1,1345 @@
+---
+layout: blogpost
+permalink: /the-specs-behind-the-specs-part-1
+title: The specs behind the specs part 1
+date: 2021-08-01
+tags: telco ASN.1 dia
+author: <a href="https://www.linkedin.com/in/sebastian-weddmark-olsson/">Sebastian Weddmark Olsson</a>, Telco newb
+---
+
+
+This will *probably* be a two piece blog post. I'll start with ASN.1
+in this very long post and then sometime later go over the Diameter
+dictionary specifications.
+
+There might be some Erlang specific paragraphs here and there, but
+this blog post is mainly about ASN.1 as a specification.
+
+# Abstract Syntax Notation version One
+
+Abstract Syntax Notation version One (ASN.1 for short) provides a
+high level description of messages. It abstracts the language
+implementations from the protocol design.
+
+It was initially used by OSI to describe email messages but are used
+by many other applications especially within telecommunications and
+cryptography.
+
+You might have heard of similar such abstract syntax notations used
+for interface definitions such as Google Protocol Buffers, or
+Facebook's Apache Thrift, but those languages have not managed by a
+standardization organ, so the owning corporations could (in theory) do
```

Comment:
but those languages have not managed by a standardization organ
->
but those languages have not been managed by a standardization organization